### PR TITLE
Fixes #24363 - Endpoint to display variables for role(s)

### PR DESCRIPTION
**Pull request by dLobatog**
There should be an endpoint at the proxy to display all variables for a
role and for many roles, so Foreman can use that to import the
variables.

### CODE REVIEW

File: lib/foreman_ansible_core/variables_extractor.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  class VariablesExtractor
+    class << self
+      def extract_variables(role_path)
+        role_files = Dir.glob("#{role_path}/**/*.yml")
+        # not anything matching item, }}, {{, ansible_hostname or 'if'
+        variables = role_files.map do |role_file|
+          candidates = File.read(role_file).scan(/{{(.*?)}}/).select do |param|
+            param.first.scan(/item/) == [] && param.first.scan(/if/) == []
+
+          end.first
```
Comment (xprazak2) [inline lib/foreman_ansible_core/variables_extractor.rb]: s/first/flatten/

File: lib/foreman_ansible_core/variables_extractor.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module ForemanAnsibleCore
+  # Implements the logic needed to read the roles and associated information
+  class VariablesExtractor
+    class << self
+      def extract_variables(role_path)
+        role_files = Dir.glob("#{role_path}/**/*.yml")
```
Comment (xprazak2) [inline lib/foreman_ansible_core/variables_extractor.rb]: Does it make sense to scan for *.yaml files as well?

But more importantly, why getting variables by scanning the files like this? I plan to test some more, but it seems fragile to me. Variables from defaults/main.yml that are used just in templates are not imported. Wouldn't it be easier just to get them from the files in defaults?

Comment (dLobatog) [reply lib/foreman_ansible_core/variables_extractor.rb]: Sometimes there is no 'defaults' folder, and we would still want to read the variables in such a case

Comment (xprazak2) [reply lib/foreman_ansible_core/variables_extractor.rb]: I do not think this is resolved. There are 3 main problems with this approach as far as I can see:

* the variables that are declared in defaults/main.yml and used in templates are not detected

* you are matching more that just variables, [filters](https://docs.ansible.com/ansible/2.5/user_guide/playbooks_filters.html#filters-for-formatting-data) are imported as variables.

* you are importing not just variables of the role, you are importing all the variables that are used in the role. Do we really want to import `ansible_distribution` as a role variable? 




What about the following:

```yaml
tasks:
    - set_fact:
        qpid_port: "{{ lookup('env', 'QPID_SERVICE_PORT') | int }}"
        qpid_host: "{{ lookup('env', 'QPID_SERVICE_HOST') }}"

    - name: "Wait for Qpid"
      wait_for:
        port: "{{ qpid_port }}"
        host: "{{ qpid_host }}"
        timeout: 600

    - name: "Check for qpid queue"
      command: "qpid-config -b {{ qpid_host }}:{{ qpid_port }} queues katello_event_queue"
      register: queue_exist
      ignore_errors: true
```

Should `qpid_port` and `qpid_host` be exported as variables?

**Comment by dLobatog** [comment]
Merging for the time being

**Pull request merged**